solvers/MotionPlanningTree.m

```diff
@@ -0,0 +1,273 @@
+classdef MotionPlanningTree
+  properties (Constant)
+    TRAPPED = 0;
+    REACHED = 1;
+    ADVANCED = 2;
+  end
+  
+  properties
+    N = 1e4; % Number of nodes for which memory should be pre-allocated
+    n = 0; % Number of nodes in the tree
+    max_edge_length = 0.1;
+    max_length_between_constraint_checks = 0.01;
+    line_color = 0.3*[1 1 1];
+    lcmgl;
+  end
+
+  methods (Abstract)
+    q = randomConfig(obj);
+    d = distanceMetric(obj, q1, q_array)
+    is_valid = isValidConfiguration(obj, q);
+    q = getVertex(obj, id);
+    q = interpolate(obj, q1, q2, interpolation_factors);
+    id_parent = getParentId(obj, id)
+  end
+
+  methods
+    function obj = init(obj, q_init)
+      obj.n = 0;
+    end
+
+    function [obj, path_ids, info] = rrtNew(obj, x_start, x_goal, options)
```

Comment:
Sorry, the new was just to prevent name clashes before I removed the `rrt` method from `MotionPlanningProblem`. I'll rename and add doc.